src/main/java/taxi/service/DriverService.java

```diff
@@ -3,4 +3,5 @@
 import taxi.model.Driver;
 
 public interface DriverService extends GenericService<Driver> {
+    Driver findByLogin(String login);
```

Comment:
Optional Driver> findByLogin(String login); - it is on the condition of the task

Reply:
you are right! 
but at the same cases with get() methods we used Optional on Dao Layer and on Service Layer unboxed it with throwing exception.